src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -657,6 +657,42 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_StoreFabricId(uint64_t f
     return err;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_GetRegulatoryLocation(uint32_t & location)
+{
+    return Impl()->ReadConfigValue(ImplClass::kConfigKey_RegulatoryLocation, location);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_StoreRegulatoryLocation(uint32_t location)
+{
+    return Impl()->WriteConfigValue(ImplClass::kConfigKey_RegulatoryLocation, location);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_GetCountryCode(char * buf, size_t bufSize, size_t & codeLen)
+{
+    return Impl()->ReadConfigValueStr(ImplClass::kConfigKey_CountryCode, buf, bufSize, codeLen);
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::_StoreCountryCode(const char * code, size_t codeLen)
```

Comment:
is country code not null terminated?

Reply:
It is, right now, ConfigurationMgr store all char * with extra length parameter.
Such as: 
inline CHIP_ERROR ConfigurationManager::StoreSerialNumber(const char * serialNum, size_t serialNumLen)